oauthenticator/oauth2.py

```diff
@@ -1136,6 +1180,18 @@ async def update_auth_model(self, auth_model):
                     auth_model["admin"] = bool(user_groups & self.admin_groups)
         return auth_model
 
+    async def _call_modify_auth_state_hook(self, auth_state):
+        """Call the modify_auth_state_hook"""
+        try:
+            auth_state = self.modify_auth_state_hook(self, auth_state)
+        except Exception as e:
+            # let hook errors raise, nothing in auth should suppress errors
+            self.log.error(f"Error in modify_auth_state_hook: {e}")
+            raise
+        if isawaitable(auth_state):
+            auth_state = await auth_state
```

Comment:
Wouldn't the exception be thrown here on the `await` than earlier? Wouldn't that require us to be inside the `try`? I could be wrong...

Reply:
You're 100% right, yes! Fixed.